dynamicArray.h

```diff
@@ -0,0 +1,100 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#define INIT_SIZE 1
+#define EXPAND_COEFFICIENT 2
+
+typedef enum {
+    SUCCESS = 0,
+    INIT_ERROR,
+    EXPAND_ERROR,
+    SHRINK_ERROR,
```

Comment:
Aren't all of these errors a type of allocation error? Can you merge these errors to a single errortype, such that it won't be ambiguous to which error you mean (since you know which function returned it)?